tests/unittests/quantizationTest.cpp

```diff
@@ -216,14 +216,161 @@ TEST_P(Quantization, end2end) {
+  //fillStableRandomData(hidden->getHandle(), 3443, 1);
+
+  hiddenInit->getPayload().zero();
+
+  Node *hidden = hiddenInit;
+
+
+  for (unsigned step = 0; step < SEQUENCE_SIZE; step++){
+    // STEP2 - Gather a single set of embeddings for the GRU
+    Node *inputEmbedded = F->createGather("gru.embedding", emb, input);
+    Node *inputSlice = F->createSlice("gru.inputSlice",
+				      inputEmbedded,
+				      {0,step,0},
+				      {BATCH_SIZE, step + 1, EMBEDDING_SIZE});
+    Node *reshape = F->createReshape("gru.reshape",
+				     inputSlice,
+				     {BATCH_SIZE, EMBEDDING_SIZE});
+  
+    // STEP3 - Generate a GRU
+    // reference implementation:
+    // https://github.com/pytorch/pytorch/blob/dd5c195646b941d3e20a72847ac48c41e272b8b2/torch/nn/_functions/rnn.py#L46
+    // similar to /examples/fr2en.cpp
+
+    auto *FCi = F->createFullyConnected("gru.fci", reshape, HIDDEN_SIZE);
+    Variable *biasI = cast<Variable>(FCi->getBias());
+    Variable *filterI = cast<Variable>(FCi->getWeights());
+    fillStableRandomData(biasI->getPayload().getHandle(), 8877, 1);
+    fillStableRandomData(filterI->getPayload().getHandle(), 1441, 1);
+
+    auto *FCh = F->createFullyConnected("gru.fch", hidden, HIDDEN_SIZE);
+    Variable *biasH = cast<Variable>(FCh->getBias());
+    Variable *filterH = cast<Variable>(FCh->getWeights());
+    fillStableRandomData(biasH->getPayload().getHandle(), 9009, 1);
+    fillStableRandomData(filterH->getPayload().getHandle(), 1001, 1);
+
+    Node *i_r = F->createSlice("gru.i_r", FCi, {0, 0},
+			       {BATCH_SIZE, EMBEDDING_SIZE});
+    Node *i_i = F->createSlice("gru.i_i", FCi, {0, EMBEDDING_SIZE},
+			       {BATCH_SIZE, 2 * EMBEDDING_SIZE});
+    Node *i_n = F->createSlice("gru.i_n", FCi, {0, 2 * EMBEDDING_SIZE},
+			       {BATCH_SIZE, 3 * EMBEDDING_SIZE});
+
+    Node *h_r = F->createSlice("gru.h_r", FCh, {0, 0},
+			       {BATCH_SIZE, EMBEDDING_SIZE});
+    Node *h_i = F->createSlice("gru.h_i", FCh, {0, EMBEDDING_SIZE},
+			       {BATCH_SIZE, 2 * EMBEDDING_SIZE});
+    Node *h_n = F->createSlice("gru.h_n", FCh, {0, 2 * EMBEDDING_SIZE},
+			       {BATCH_SIZE, 3 * EMBEDDING_SIZE});
+
+    Node *resetgate = F->createSigmoid("gru.resetgate",
+				       F->createAdd("i_r_plus_h_r", i_r, h_r));
+    Node *inputgate = F->createSigmoid("gru.inputgate",
+				       F->createAdd("i_i_plus_h_i", i_i, h_i));
+    Node *newgate = F->createTanh("gru.newgate",
+				  F->createAdd("i_n_plus_rg_mult_h_n", i_n,
+					       F->createMul("rg_mult_h_n", resetgate, h_n)));
+    hidden = F->createAdd("gru.newhidden", newgate,
+			  F->createMul("ig_mult_hmng", inputgate,
+				       F->createSub("hidden_minus_newgate", hidden, newgate)));
+
+    
+  }
+  // No-op TopK selection to test quantization
+  Node *downsample = F->createTopK("gru.downsample", hidden, EMBEDDING_SIZE/2);
+  
+  SaveNode *SN = F -> createSave("save", {downsample,0});
+  return {F, SN};
+}
+
+TEST_P(Quantization, end2endGRU) {
```

Comment:
how long does it take to run the test?

Reply:
when you run a coverage test, do you see quantization operator conversion code be fully covered now?